components/list/list-item-nav-mixin.js

```diff
@@ -0,0 +1,112 @@
+import '../colors/colors.js';
+import { css } from 'lit';
+import { ListItemButtonMixin } from './list-item-button-mixin.js';
+
+export const ListItemNavMixin = superclass => class extends ListItemButtonMixin(superclass) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether the list item is the current page in a navigation context
+			 * @type {boolean}
+			 */
+			current: { type: Boolean, reflect: true },
+			_childCurrent: { type: Boolean, reflect: true, attribute: '_child-current' },
+		};
+	}
+
+	static get styles() {
+
+		const styles = [ css`
+			:host(:not([button-disabled])) {
+				--d2l-list-item-content-text-color: var(--d2l-color-ferrite);
+			}
+			.d2l-list-item-content ::slotted(*) {
+				width: 100%;
+			}
+			:host([current]) [slot="outside-control-container"] {
+				border: 3px solid var(--d2l-color-celestine);
+				margin-block: -1px;
+			}
+			:host([_focusing-primary-action]:not([current])) [slot="outside-control-container"] {
+				border: 2px solid var(--d2l-color-celestine);
+			}
+			:host([current]) [slot="control-container"]::before,
+			:host([current]) [slot="control-container"]::after {
+				border-color: transparent;
+			}
+			:host([_focusing-primary-action]) .d2l-list-item-content {
+				--d2l-list-item-content-text-outline: none;
+			}
+			:host([_hovering-primary-action]) .d2l-list-item-content,
+			:host([_focusing-primary-action]) .d2l-list-item-content {
+				--d2l-list-item-content-text-color: var(--d2l-color-ferrite);
+				--d2l-list-item-content-text-decoration: none;
+			}
+
+		` ];
+
+		super.styles && styles.unshift(super.styles);
+		return styles;
+	}
+
+	constructor() {
+		super();
+		this.current = false;
+		this._childCurrent = false;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.addEventListener('d2l-list-item-nav-set-child-current', async(e) => {
+			await this.updateComplete; // ensure button exists
+			if (e.target === this || !this._button) return;
+			this._childCurrent = e.detail.value;
+		});
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (this.current) {
+			this.#setAriaCurrent('page');
+			this._dispatchSetChildCurrentEvent(true);
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (changedProperties.has('current') || changedProperties.has('_childCurrent')) {
```

Comment:
Is there a way we can set it up so we avoid extra render cycles? - currently it looks like this will render, and then will re-render as result of updating `_childCurrent`. Is it in `updated` because `#setAriaCurrent` depends on `_button` (updating `ariaCurrent` outside of render)?

Reply:
I believe the setting of `_childCurrent` was not necessary here since it's also set in `_onButtonClick` and also as a result of `d2l-list-item-nav-set-child-current` but it's possible there's a scenario I'm not thinking of.